src/Nova/Role.php

```diff
@@ -88,11 +98,15 @@ public function fields(Request $request)
                         'label'  => __($permission->name),
                     ];
                 })
-                ->groupBy('group')
-                ->toArray()),
+                    ->groupBy('group')
+                    ->toArray()),
 
             Text::make(__('Users'), function () {
-                return $this->users()->count();
+                $expirationTime = config('permission.cache.nova_expiration_time', now()->addMinute());
+
+                return cache()->remember('sereny-nova-permissions-user-count', $expirationTime, function () {
+                    return $this->users()->count();
+                });
```

Comment:
This will show the same count for every row.
Besides, if you add a User to a Role, the count will not change for 24h:
 ```PHP
     'expiration_time' => DateInterval::createFromDateString('24 hours'),
 ```
The eager load count I sent in my last change does not load the users as you pointed out in the other solution. It is a very light eager query with a even lighter memory footprint.

Please take a look at https://laravel.com/docs/11.x/eloquent-relationships#relationship-counting-and-custom-select-statements